apis/operator/v1alpha1/apigateway_types.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type State string
+
+const (
+	Ready      State = "Ready"
+	Processing State = "Processing"
+	Error      State = "Error"
+	Deleting   State = "Deleting"
+	Warning    State = "Warning"
+)
+
+// APIGatewaySpec defines the desired state of APIGateway
+type APIGatewaySpec struct {
+}
+
+// APIGatewayStatus defines the observed state of APIGateway
+type APIGatewayStatus struct {
+	// State signifies current state of APIGateway. Value can be one of ("Ready", "Processing", "Error", "Deleting", "Warning").
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=Processing;Deleting;Ready;Error;Warning
+	State State `json:"state"`
+	// Description of APIGateway status
+	Description string `json:"description,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
```

Comment:
We should discuss if having the resource Cluster scoped isn't a better option
```suggestion
//+kubebuilder:printcolumn:JSONPath=".status.state",name="State",type="string"
//+kubebuilder:resource:scope=Cluster

```